### PR TITLE
Update pubsub samples

### PR DESCRIPTION
**Pull request by jonparrott**
/cc @lukesneeringer

### CODE REVIEW

**Comment by jonparrott** [comment]
I'm gonna go ahead and merge because docs are broken at the moment. Will address any issues/concerns in a follow-up PR.

File: pubsub/cloud-client/subscriber.py [hunk 1]
```diff
@@ -22,70 +22,91 @@
...
+        print('Received message: {}'.format(message))
+        message.ack()
 
-    print('Received {} messages.'.format(len(results)))
+    subscriber.subscribe(subscription_path, callback=callback)
 
-    for ack_id, message in results:
-        print('* {}: {}, {}'.format(
-            message.message_id, message.data, message.attributes))
+    # The subscriber is non-blocking, so we must keep the main thread from
```
Comment (andrewsg) [inline pubsub/cloud-client/subscriber.py]: Is there any deliberately blocking method in the new API, similar to line 68 of the old sample?

Comment (jonparrott) [reply pubsub/cloud-client/subscriber.py]: No, but see https://github.com/GoogleCloudPlatform/google-cloud-python/issues/3888

File: pubsub/cloud-client/publisher.py [hunk 2]
```diff
@@ -23,55 +23,100 @@
...
+    for n in range(1, 10):
+        data = u'Message number {}'.format(n)
+        # Data must be a bytestring
+        data = data.encode('utf-8')
+        message_future = publisher.publish(topic_path, data=data)
+        futures.append(message_future)
+
+    print('Published message IDs:')
+    for future in futures:
+        print(future.result())
```
[the remark this answers is not in the document]
Comment (jonparrott) [reply pubsub/cloud-client/publisher.py]: Done in 254438b

**Pull request merged**